src/modules/theme/hocs/withTheme.js

```diff
@@ -0,0 +1,32 @@
+// @flow
+
+import * as React from 'react'
+import { wrapDisplayName } from 'recompose'
+
+import type { ThemeType } from '../constants/theme'
+import { brightColors } from '../constants/colors'
+import dimensions from '../constants/dimensions'
+import { arabicFonts, defaultFonts } from '../constants/fonts'
+
+type MapPropsToLanguageType<Props> = Props => string
+
+const ARABIC_LANGUAGES = ['ar', 'fa', 'ku']
+
+const withTheme = <Props> (mapPropsToLanguage?: MapPropsToLanguageType<Props>) =>
+  (Component: React.ComponentType<Props>): React.ComponentType<$Diff<Props, {| theme: ThemeType |}>> => {
+    return class extends React.Component<{}> {
+      static displayName = wrapDisplayName(Component, 'withTheme')
+
+      render () {
+        const language = mapPropsToLanguage && mapPropsToLanguage(this.props)
+        const theme: ThemeType = {
+          colors: brightColors,
```

Comment:
Components which use this HOC will not be 100% testable because the theme is not mockable.

The "right" way to do this would be to add the "theme" to the context. This HOC would get the theme from the context then and not by importing it.